bin/scripts/fix_broken_ratings.js

```diff
@@ -0,0 +1,100 @@
+                                did: fixed_device_id,
+                                uid: brokenUser.uid
+                            };
+                            // map correct uid and did values to drill events records
+                            countlyDrillDB.collection("drill_events" + crypto.createHash('sha1').update("[CLY]_star_rating" + app._id).digest('hex'))
+                                .updateMany({did: brokenUser.did}, {
+                                    $set: {
+                                        did: correctReference.did,
+                                        uid: correctReference.uid
+                                    }
+                                }, function(err) {
+                                    if (!err) {
+                                        countlyDb.collection('app_users' + app._id).remove({_id: brokenUser._id}, function(err) {
+                                            if (!err) {
+                                                countlyDb.collection('app_users' + app._id).findOne({did: fixed_device_id}, function(err, correctAppUser) {
+                                                    if (!err && !correctAppUser) {
+                                                        brokenUser._id = crypto.createHash('sha1')
+                                                            .update(app.app_key + fixed_device_id + "")
+                                                            .digest('hex');
+                                                        brokenUser.did = fixed_device_id;
+                                                        countlyDb.collection('app_users' + app._id).insert(brokenUser, function(err) {
+                                                            if (!err) {
+                                                                // map correct uid and did values to broken feedback records
+                                                                countlyDb.collection("feedback" + app._id)
+                                                                    .updateMany({device_id: brokenUser.did}, {
+                                                                        $set: {
+                                                                            device_id: brokenUser.did,
+                                                                            uid: brokenUser.uid
+                                                                        }
+                                                                    }, function(err) {
+                                                                        if(!err) doneUser();
+                                                                    });
+                                                            } else console.log(err);
+                                                        });
+                                                    } else {
```

Comment:
We can enter this state when there is error, but correctAppUser still does not exist, and script will crash when accessing correctAppUser properties